stan/math/prim/prob/wiener4_lcdf.hpp

```
@@ -0,0 +1,745 @@
+  const auto w = 1 - wn;
+
+  const auto K1 = 0.5 * (fabs(v) / a * y - w);
+  const auto arg
+      = fmax(0, fmin(1, exp(v * a * w + square(v) * y / 2 + err) / 2));
+  const auto K2 = (arg == 0) ? INFTY
+                             : (arg == 1) ? NEGATIVE_INFTY
+                                          : -sqrt(y) / 2 / a * inv_Phi(arg);
+  const auto K_small_value = ceil(fmax(K1, K1 + K2));
+
+  const auto api = a / pi();
+  const auto v_square = square(v);
+  const auto sqrtL1 = sqrt(1 / y) * api;
+  const auto sqrtL2 = sqrt(
+      fmax(1.0, -2 / y * square(api)
+                    * (err + log(pi() * y / 2 * (v_square + square(pi() / a)))
+                       + v * a * w + v_square * y / 2)));
+  const auto K_large_value = ceil(fmax(sqrtL1, sqrtL2));
+
+  auto lg = LOG_TWO + LOG_PI - 2.0 * log(a);
+
+  if (3 * K_small_value < K_large_value) {
+    const auto sqrt_y = sqrt(y);
+    const auto vy = v * y;
+    auto ans = ret_t(0.0);
+    ret_t fplus = NEGATIVE_INFTY;
+    ret_t fminus = NEGATIVE_INFTY;
+    for (auto k = K_small_value; k >= 0; k--) {
+      auto rj = a * (2 * k + w);
+      auto dj = std_normal_lpdf(rj / sqrt_y);
+      auto pos1 = dj + logMill((rj - vy) / sqrt_y);
+      auto pos2 = dj + logMill((rj + vy) / sqrt_y);
+      fplus = log_sum_exp(fplus, log_sum_exp(pos1, pos2));
+      rj = a * (2 * k + 2 - w);
+      dj = std_normal_lpdf(rj / sqrt_y);
+      auto neg1 = dj + logMill((rj - vy) / sqrt_y);
+      auto neg2 = dj + logMill((rj + vy) / sqrt_y);
+      fminus = log_sum_exp(fminus, log_sum_exp(neg1, neg2));
+    }
+    if (fplus > fminus) {
+      ans = log_diff_exp(fplus, fminus);
+    } else if (fplus < fminus) {
+      ans = log_diff_exp(fminus, fplus);
+    } else {
+      ans = NEGATIVE_INFTY;
+    }
+    ret_t log_distribution = ans + -v * a * w - square(v) * y / 2;
+    return NaturalScale ? exp(log_distribution) : log_distribution;
+  } else {
+    const auto log_a = log(a);
+    const auto log_v = log(fabs(v));
```

Question:  Is this a case where `v` might be legitimately negative?  If not, let it fail rather than applying `fabs`.

---

`v` is the drift rate and is allowed to be positive and negative.